src/libraries/System.Threading.Tasks.Dataflow/tests/Dataflow/TransformManyBlockTests.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,663 @@
+
+            using (tb.LinkTo(tb))
+            {
+                tb.Post(0);
+                await tcs.Task;
+                tb.Complete();
+            }
+        }
+
+        [Fact]
+        public async Task TestProducerConsumerAsyncEnumerable()
+        {
+            foreach (TaskScheduler scheduler in new[] { TaskScheduler.Default, new ConcurrentExclusiveSchedulerPair().ConcurrentScheduler })
+                foreach (int maxMessagesPerTask in new[] { DataflowBlockOptions.Unbounded, 1, 2 })
+                    foreach (int boundedCapacity in new[] { DataflowBlockOptions.Unbounded, 1, 2 })
+                        foreach (int dop in new[] { 1, 2 })
+                            foreach (int elementsPerItem in new[] { 1, 3, 5 })
+                            {
+                                const int Messages = 50;
+                                var options = new ExecutionDataflowBlockOptions
+                                {
+                                    BoundedCapacity = boundedCapacity,
+                                    MaxDegreeOfParallelism = dop,
+                                    MaxMessagesPerTask = maxMessagesPerTask,
+                                    TaskScheduler = scheduler
+                                };
+                                TransformManyBlock<int, int> tb = new TransformManyBlock<int, int>(i => AsyncEnumerable.Repeat(i, elementsPerItem), options);
+
+                                await Task.WhenAll(
+                                    Task.Run(async delegate
+                                    { // consumer
+                                        int i = 0;
+                                        int processed = 0;
+                                        while (await tb.OutputAvailableAsync())
+                                        {
+                                            Assert.Equal(expected: i, actual: await tb.ReceiveAsync());
+                                            processed++;
+                                            if (processed % elementsPerItem == 0)
+                                            {
+                                                i++;
+                                            }
+                                        }
+                                    }),
+                                    Task.Run(async delegate
+                                    { // producer
+                                        for (int i = 0; i < Messages; i++)
+                                        {
+                                            await tb.SendAsync(i);
+                                        }
+                                        tb.Complete();
+                                    }));
+                            }
+        }
+
+        [Fact]
+        public async Task TestMessagePostponementAsyncEnumerable()
+        {
+            const int Excess = 10;
```

Comment:
```suggestion
            const int excess = 10;
```
private constants should be written in lower case